hudi-spark-datasource/hudi-spark2/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/MORBootstrap24FileFormat.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.parquet
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hudi.{HoodieTableSchema, HoodieTableState}
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.sql.BootstrapMORIteratorFactory.MORBootstrapFileFormat
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.execution.datasources.PartitionedFile
+import org.apache.spark.sql.sources.Filter
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{BootstrapMORIteratorFactory, SparkSession}
+
+class MORBootstrap24FileFormat(shouldAppendPartitionValues: Boolean,
```

Comment:
Could we make the new file format implementation independent of existing implementation and directly  extending `ParquetFileFormat`?  That way, we can keep the current file format implementation intact and remove current file format classes easily in the future.

For simplicity, not all features need to be supported in the new file format, such as full schema evolution (schema-on-read).